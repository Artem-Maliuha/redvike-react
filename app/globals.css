@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --primary: #1A1A1E;
  --background-rgb: 255, 255, 255;
}

body {
  color: rgb(var(--primary));
  background: rgb(var(--background-rgb));
}
