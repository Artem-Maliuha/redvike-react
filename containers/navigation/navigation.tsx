import Link from 'next/link'

interface NavigationProps {
  title?: string
}

export const Navigation = (props: NavigationProps) => {
  const { title } = props

  if (!title) {
    return <h1 className="text-xl font-bold">Pokedex</h1>
  }

  return (
    <nav className="text-xl font-bold">
      <Link href="/">Home</Link>
      {' / '}
      <span className="capitalize">{title}</span>
    </nav>
  )
}
