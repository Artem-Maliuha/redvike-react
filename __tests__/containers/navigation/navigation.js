import renderer from 'react-test-renderer'
import { describe, it, expect } from '@jest/globals'
import { Navigation } from '../../../containers/navigation'

describe('<Navigation />', () => {
  it('snapshot', () => {
    const component = renderer.create(<Navigation />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('snapshot with title', () => {
    const component = renderer.create(<Navigation title="Pokemon" />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
