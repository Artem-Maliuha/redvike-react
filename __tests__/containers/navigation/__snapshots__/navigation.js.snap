// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> snapshot 1`] = `
<h1
  className="text-xl font-bold"
>
  Pokedex
</h1>
`;

exports[`<Navigation /> snapshot with title 1`] = `
<nav
  className="text-xl font-bold"
>
  <a
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    Home
  </a>
   / 
  <span
    className="capitalize"
  >
    Pokemon
  </span>
</nav>
`;
